qemu/tests/qemu_guest_agent.py

```diff
@@ -355,6 +355,7 @@ def __gagent_check_shutdown(self, shutdown_mode):
         self.gagent.shutdown(shutdown_mode)
 
     def __gagent_check_serial_output(self, pattern):
+        self.vm.create_serial_console()
```

Comment:
if you created self.vm before, `create_serial_console()` should be called as well.

Reply:
I found something interesting. Due to our auto bug 1516187, a timeout always occurs when staf attempts to login vm via ssh, and then serial console will be used to login vm. At that time, `vm.create_serial_console` will be called again in `wait_for_login` and session will be **closed** when login method done.
`            session = self.wait_for_serial_login(timeout, internal_timeout,
                                                 restart_network, username,
                                                 password, False)
            if serial:
                return session
            session.close()        # close the session`
I guess we need to do some operations to avoid this situation.

Reply:
When logging vm via serial console, the code will judge if serial console exists. Because serial console has been created when create vm, vm.create_serial_console will not be called in this stage.

Reply:
if not (console and os.path.exists(console.inpipe_filename)):
            create_console = getattr(self, "create_%s" % console_type)
            create_console()
            console = getattr(self, console_type)

Reply:
Yes, as I told you, I can confirm that this case can PASS when login with ssh, and FAIL when login with serial console in my test environment every time. And accords to your said, now I have no idea why this happened.